### PR TITLE
added feign client to push files to azure ftp

### PR DESCRIPTION
**Pull request by kurt-uk**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PRO-4675

### Change description ###
added feign client to push files to azure ftp, added signature to yml that has to be replaced locally to work. modified solution to add grants for day to file rather than a grant per file. adjusted excela and ironMountian to reflect this. Added scheduler annotation to notification config that was removed by a different commit.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -3,7 +3,12 @@ server:
 
 resttemplate:
   httpclient:
-    timeout:10000
+    timeout: 10000
+
+ftp.client:
+  url: https://green.blob.core.windows.net
```
Comment (sanjayparekh-hmcts) [inline src/main/resources/application.yml:9]: does this need to be in vault too?

File: src/main/java/uk/gov/hmcts/probate/service/FileTransferService.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import java.nio.file.Files;
+
+@Slf4j
+public class FileTransferService {
+
+    private final FileTransferApi fileTransferApi;
+    private static final String VALID_FROM = "2018-03-28";
+    private static final String SS = "b";
+    private static final String SRT = "sco";
+    private static final String SP = "rwdlac";
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/service/FileTransferService.java:20]: do we know what these codes are for? perhaps we can use a more descriptive name for them

File: src/main/java/uk/gov/hmcts/probate/service/NotificationService.java [hunk 3]
```diff
@@ -144,13 +145,13 @@ private Document getGeneratedSentEmailDocument(SendEmailResponse response, Strin
         return personalisation;
     }
 
-    private Map<String, String> getExcelaPersonalisation(Long id, CaseData caseData) {
+    private Map<String, String> getExcelaPersonalisation(List<ReturnedCaseDetails> cases) {
         HashMap<String, String> personalisation = new HashMap<>();
 
+        StringBuilder data = getBuiltData(cases);
+
         personalisation.put("excelaName", LocalDateTime.now().format(EXCELA_DATE) + "will");
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/service/NotificationService.java]: probably want these personalisation keys as final const String too

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yml (hunk 1 above)
Comment (kurt-uk) [reply src/main/resources/application.yml:9]: apparently not, i asked them what we would need to keep private, no one can do anything as long as they dont have the signature, 
however if you feel i should put them there then i can do that too!

Comment (sanjayparekh-hmcts) [reply src/main/resources/application.yml:9]: Nope!

File: src/main/java/uk/gov/hmcts/probate/service/FileTransferService.java (hunk 2 above)
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/probate/service/FileTransferService.java:20]: not a single clue, i asked James what they were for and it seems that they dont know either, i apparently just need them, which i do. however i tried to include some descriptive names from what i interpreted such as the valid from date which should have been `sv`. unfortunately they couldnt inform me of their purpose so i dont know what to call them other than their paramter value

File: src/main/java/uk/gov/hmcts/probate/service/NotificationService.java (hunk 3 above)
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/probate/service/NotificationService.java]: sure thing

**Review by RebeccaBaker — APPROVED** [review]

**Pull request merged**